### PR TITLE
[ticket/15328] Disable checkbox if notification method isn't supported

### PR DESCRIPTION
**Pull request by kasimi**
Stuffing the instances into the `type` and `method` data arrays in `manager.php` doesn't feel like an elegant solution but it's the best I could find that is extensible.

As an alternative, it could all be done in the UCP module, probably with some `instanceof` checks, but that would make it deal with things that it shouldn't deal with, in my opinion. Other suggestions are welcome.

Checklist:

- [x] Correct branch: master for new features; 3.2.x, 3.1.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master / 3.2.x](https://area51.phpbb.com/docs/master/coding-guidelines.html), [3.1.x](https://area51.phpbb.com/docs/31x/coding-guidelines.html)
- [x] Commit follows commit message [format](https://wiki.phpbb.com/Git#Commit_Messages)

https://tracker.phpbb.com/browse/PHPBB3-15328

### CODE REVIEW

**Comment by kasimi** [comment]
Any idea what Travis is trying to tell me?

File: phpBB/phpbb/notification/method/email.php [hunk 1]
```diff
@@ -56,10 +56,14 @@ public function get_type()
 	/**
 	* Is this method available for the user?
 	* This is checked on the notifications options
+	*
+	* @param \phpbb\notification\type\type_interface $notification_type An optional instance of a notification type. If provided, this
+	*																	method additionally checks if the type provides an email template.
+	* @return bool
 	*/
-	public function is_available()
+	public function is_available(\phpbb\notification\type\type_interface $notification_type = null)
```
Comment (CHItA) [inline phpBB/phpbb/notification/method/email.php]: Add a `use` statement maybe.

File: phpBB/phpbb/notification/method/jabber.php [hunk 2]
```diff
@@ -56,10 +56,14 @@ public function get_type()
 	/**
 	* Is this method available for the user?
 	* This is checked on the notifications options
+	*
+	* @param \phpbb\notification\type\type_interface $notification_type	An optional instance of a notification type. If provided, this
+	*																	method additionally checks if the type provides an email template.
+	* @return bool
 	*/
-	public function is_available()
+	public function is_available(\phpbb\notification\type\type_interface $notification_type = null)
```
Comment (CHItA) [inline phpBB/phpbb/notification/method/jabber.php]: Same as above.

**Comment by kasimi** [comment]
At the moment it's a front-end fix. Form values aren't checked. Guaranteeing consistent user settings in the database would require a migration to uncheck all existing wrong values, too.

File: phpBB/styles/prosilver/template/ucp_notifications.html [hunk 3]
```diff
@@ -31,7 +31,7 @@
 								<!-- IF notification_types.EXPLAIN --><br />&nbsp; &nbsp;{notification_types.EXPLAIN}<!-- ENDIF -->
 							</td>
 							<!-- BEGIN notification_methods -->
-								<td class="mark"><input type="checkbox" name="{notification_types.TYPE}_{notification_methods.METHOD}"<!-- IF notification_methods.SUBSCRIBED --> checked="checked"<!-- ENDIF --> /></td>
+								<td class="mark"><input type="checkbox" name="{notification_types.TYPE}_{notification_types.notification_methods.METHOD}"<!-- IF notification_types.notification_methods.AVAILABLE and notification_types.notification_methods.SUBSCRIBED --> checked="checked"<!-- ENDIF --><!-- IF not notification_types.notification_methods.AVAILABLE --> disabled="disabled"<!-- ENDIF --> /></td>
```
Comment (Elsensee) [inline phpBB/styles/prosilver/template/ucp_notifications.html]: I believe it must be `notification_methods.AVAILABLE" only? Or was it just wrong before?

Comment (kasimi) [reply phpBB/styles/prosilver/template/ucp_notifications.html]: It used the values of the header row before, which worked coincidentally. `notification_types.notification_methods.METHOD` is the actual row's value. https://wiki.phpbb.com/Template.assign_block_vars#Example_.232_Nested_blocks

Comment (Elsensee) [reply phpBB/styles/prosilver/template/ucp_notifications.html]: Ah, that makes sense!

**Pull request merged**